CMakeLists.txt

```diff
@@ -460,7 +463,7 @@ if(WITH_MGR)
 endif()
 
 # require minimally the bundled version
-find_package(Boost 1.61 COMPONENTS thread system regex random program_options date_time iostreams REQUIRED)
+find_package(Boost 1.61 COMPONENTS ${BOOST_COMPONENTS} REQUIRED)
```

Comment:
it comes from a few lines above https://github.com/ceph/ceph/pull/12143/commits/3eef10b977294f6d2d16f3fb6e61a99a35cbc6df#diff-af3b638bc2a3e6c650974192a53c7291R459. It was there but never used. I assume the intention was to pass it to find_package so I went ahead and did that. Its unrelated to this PR however.